exp-player/addon/components/exp-card-sort/component.js

```diff
@@ -136,26 +136,28 @@ export default ExpFrameBaseComponent.extend({
   // Represent the sorted cards in a human-readable format for storage in the database
   responses: Ember.computed(function() {
       // Final data should be returned as {
-      //    cardSort1: object {categoryName: [cardIdentifiers] }
-      //    cardSort2: object {categoryName: [cardIdentifiers] }
+      //    cardSort1: object {cardId: categoryId,...}
+      //    cardSort2: object {cardId: categoryId,...}
       // }
       let responses = {};
       let cardSortResponse = this.get('cardSortResponse');
       if (cardSortResponse) {
           responses['cardSort1'] = {};
-          for (let category of cardSortResponse) {
-              let name = category.name.split('.').pop();
-              responses['cardSort1'][name] = category.cards.map((cardItem) => cardItem.id);
+          for (var i=0; i < cardSortResponse.length; i++) {
+              for (let card of cardSortResponse[i].cards) {
+                  responses['cardSort1'][card.id] = i + 1;
+              }
           }
       }
       if (this.get('page') === 'cardSort2') {
           cardSortResponse = this.get('buckets2');
           responses['cardSort2'] = {};
           // Assumption: this unpacks a list of { categories: {name: name, cards: [cards]} } objects
           for (let categorySet of cardSortResponse) {
-              for (let category of categorySet.categories) {
-                  let name = category.name.split('.').pop();
-                  responses['cardSort2'][name] = category.cards.map((cardItem) => cardItem.id);
+              for (var j=0; j < categorySet.categories.length; j++) {
```

Comment:
Can you please update the source code comments in ways that will make this code more maintainable? (or at least to not be inaccurate)

Even if not a full YUIDoc comment, future developers who work with this code should not have to reverse-engineer any mystery data structures to understand what a block of code does / returns.